compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -144,24 +144,29 @@ impl DepKind {
 
 macro_rules! define_dep_nodes {
     (
-     $($(#[$attr:meta])*
-        [$($modifiers:tt)*] fn $variant:ident($($K:tt)*) -> $V:ty,)*) => {
+     $( $(#[$builtin_attr:meta])* $builtin_node:ident ),+ $(,)?
+     ;
+     $($(#[$query_attr:meta])*
+        [$($modifiers:tt)*] fn $query:ident($($K:tt)*) -> $V:ty,)*) => {
 
         #[macro_export]
         macro_rules! make_dep_kind_array {
-            ($mod:ident) => {[ $($mod::$variant()),* ]};
+            ($mod:ident with_builtin_nodes) => {[ $($mod::$query(),)* $($mod::$builtin_node()),+ ]};
+            ($mod:ident) => {[ $($mod::$query()),* ]};
```

Comment:
Instead of having 2 versions of this macro and singling out builtin nodes, should we provide dummy implementations of `QueryStruct` for those builtin nodes?